### PR TITLE
Standardize err not found

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR standardizes on `model.IsErrNotFound` for checking if an error was one of the `not found` variety.  Previously there was a mix of `store.IsErrNotFound` and `errors.Is(err, sql.ErrNoRows)` and another pattern for plugin API calls.

This simplifies error checking and reduces mistakes.  For example cases where the arguments were reversed for `errors.Is(err, sql.ErrNoRows)` have been fixed.

#### Ticket Link
Follow-up to https://github.com/mattermost/focalboard/issues/2750

### CODE REVIEW

**Review by Pinjasaur — APPROVED** [review]
LGTM, nice refactor

File: server/services/store/storetests/boards.go [hunk 1]
```diff
@@ -106,7 +105,7 @@ func testGetBoard(t *testing.T, store store.Store) {
 
 	t.Run("nonexisting board", func(t *testing.T) {
 		rBoard, err := store.GetBoard("nonexistent-id")
-		require.ErrorIs(t, err, sql.ErrNoRows)
+		require.True(t, model.IsErrNotFound(err))
```
Comment (jespino) [inline server/services/store/storetests/boards.go:108]: nit: A better error message would be great if not you get something like:

```
=== RUN   TestSQLStore/BoardStore/GetMemberForBoard/should_return_a_no_rows_error_for_nonexisting_membership
    boards.go:571: 
        	Error Trace:	boards.go:571
        	Error:      	Should be false
        	Test:       	TestSQLStore/BoardStore/GetMemberForBoard/should_return_a_no_rows_error_for_nonexisting_membership
```

**Review by jespino — APPROVED** [review]
LGTM, I added a comment that can be applied in multiple places, but other than that, it looks great.

File: server/services/store/storetests/boards.go (hunk 1 above)
Comment (wiggin77) [reply server/services/store/storetests/boards.go]: Added descriptions to the asserts.  https://github.com/mattermost/focalboard/pull/2834/commits/4fb77221e7f68c213be6de5865bb318dd20f2618

**Pull request merged**